gemrb/core/GUI/GameControl.cpp

```diff
@@ -1937,6 +1937,12 @@ static std::map<std::string, std::vector<std::string>> pstWMapExits {
 // has to be a plain travel region and on the whitelist
 bool GameControl::ShouldTriggerWorldMap(const Actor *pc) const
 {
+
+	//if we havent reached the key area, we need return false to allow regular travel instead
+	Scriptable* Sender = (Scriptable*)pc;
+	bool keyAreaVisited = core->HasFeature(GF_TEAM_MOVEMENT) && CheckVariable(Sender, "AR0500_Visited", "GLOBAL") == 1;
+	if (!keyAreaVisited) return false;
+
 	if (!core->HasFeature(GF_TEAM_MOVEMENT)) return false;
```

Comment:
Purely from the code perspective: Can you drop the 2nd check for `GF_TEAM_MOVEMENT` (always true then), or just move it up and skip yours?

Reply:
I was about to point this out myself. let's fix that.

Reply:
No need for Sender either, but let me take a broader look first.

Reply:
Oh yeah, I didn't even notice that! Whoops...

I also wasn't sure why I needed a sender, it just seemed like most instances of CheckVariable also include that. Sometimes it's the 'map' rather than an actor. I didn't ask any questions...

It does work in game though, so this must at least be 15% right :)